Elisa Aguirre Arias

2/12/26

738894

In [2]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import Ridge
from sklearn.linear_model import Lasso
from scipy import stats
import statsmodels.api as sm

In [3]:
df= pd.read_excel(r"C:\Users\elisa\Downloads\aguasfeo.xlsx")

In [4]:
df.columns = df.columns.str.strip()
df

Tamaño Hielo     Sabor  Score
0  grande  sin     kahlúa    8.2
1  grande  con     kahlúa    9.0
2   chica  sin     kahlúa    7.0
3   chica  con     kahlúa    7.5
4  grande  con   horchata   10.0
5   chica  sin   horchata    6.0
6   chica  con   horchata    6.4
7  grande  sin   horchata    9.5

**Regresión lineal**

In [5]:
df1 = pd.DataFrame({
    "Tamaño": (df["Tamaño"] == "grande").astype(int),
    "Hielo": (df["Hielo"] == "con ").astype(int),
    "Kahlúa": (df["Sabor"] == "kahlúa").astype(int),
    "Horchata": (df["Sabor"] == "horchata").astype(int)
})

df1.head(10)


Tamaño  Hielo  Kahlúa  Horchata
0       1      0       1         0
1       1      1       1         0
2       0      0       1         0
3       0      1       1         0
4       1      1       0         1
5       0      0       0         1
6       0      1       0         1
7       1      0       0         1

In [6]:
x1 = df1[["Tamaño","Hielo","Kahlúa","Horchata"]]
x = sm.add_constant(x1)
y = df[["Score"]].values.reshape(-1,1)

In [7]:
xd = sm.add_constant(x)
model = sm.OLS(y,xd)
results = model.fit()
results.summary()

C:\Users\elisa\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Thu, 12 Feb 2026   Prob (F-statistic):             0.0473
Time:                        13:31:30   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          4.3000      0.320     13.417      0.000       3.410       5.190
Tamaño         2.4500      0.555      4.414      0.012       0.909       3.991
Hielo          0.5500      0.555      0.991      0.378      -0.991       2.091
Kahlúa         2.1250      0.320      6.631      0.003       1.235       3.015
Horchata       2.1750      0.320      6.787      0.002       1.285       3.065
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     3.17e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.64e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**Parte 2**

sin hielo=-1 con=1

grande=1 chica=-1

In [16]:
df2 = pd.DataFrame({
    "Tamaño": ((df["Tamaño"] == "chica").astype(int) * 2 - 1),
    "Hielo": ((df["Hielo"] == "con ").astype(int) * 2 - 1),
    "Kahlúa": (df["Sabor"] == "kahlúa").astype(int),
    "Horchata": (df["Sabor"] == "horchata").astype(int)
})
df2.head(10)


Tamaño  Hielo  Kahlúa  Horchata
0      -1     -1       1         0
1      -1      1       1         0
2       1     -1       1         0
3       1      1       1         0
4      -1      1       0         1
5       1     -1       0         1
6       1      1       0         1
7      -1     -1       0         1

In [17]:
X2 = sm.add_constant(df2)
y2 = df["Score"]

In [18]:
modelo2 = sm.OLS(y2, X2)
resultados2 = modelo2.fit()
resultados2.summary()

C:\Users\elisa\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=8 observations were given.
  return hypotest_fun_in(*args, **kwds)


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Score   R-squared:                       0.837
Model:                            OLS   Adj. R-squared:                  0.714
Method:                 Least Squares   F-statistic:                     6.824
Date:                Thu, 12 Feb 2026   Prob (F-statistic):             0.0473
Time:                        13:39:45   Log-Likelihood:                -6.6425
No. Observations:                   8   AIC:                             21.29
Df Residuals:                       4   BIC:                             21.60
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.3000      0.185     28.644      0.000       4.786       5.814
Tamaño        -1.2250      0.278     -4.414      0.012      -1.996      -0.454
Hielo          0.2750      0.278      0.991      0.378      -0.496       1.046
Kahlúa         2.6250      0.293      8.973      0.001       1.813       3.437
Horchata       2.6750      0.293      9.143      0.001       1.863       3.487
==============================================================================
Omnibus:                        7.172   Durbin-Watson:                   1.431
Prob(Omnibus):                  0.028   Jarque-Bera (JB):                1.242
Skew:                          -0.048   Prob(JB):                        0.537
Kurtosis:                       1.072   Cond. No.                     2.70e+16
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The smallest eigenvalue is 1.64e-32. This might indicate that there are
strong multicollinearity problems or that the design matrix is singular.
"""

**conclusion**

En las dos regresiones el R² es de 83.7%, lo que significa que el modelo explica el 83.7% del comportamiento del score. Cuando cambiamos la codificación de (0,1) a (-1,1), los coeficientes sí cambian, pero el ajuste del modelo no cambia. Es decir, sigue siendo el mismo modelo, solo que cambia la forma en la que se interpretan los valores.